### PR TITLE
[MNT] Further isolation of datatypes

### PR DESCRIPTION
**Pull request by TonyBagnall**
see [#42](https://github.com/aeon-toolkit/aeon/issues/42), #974

Shifts some testing functionality out of the soon to be deprecated datatypes module, remove references to scitype and mtype

1. ALL_TIME_SERIES_TYPES has moved to base init from from aeon.datatypes import ALL_TIME_SERIES_TYPES. Not sure thats its final place
2 aeon.datatypes._utilities move to utils
3. use of from aeon.datatypes._panel._convert import from_nested_to_multi_index changed to collections
4. mtypes variable names changed
5. datatypes/examples moved to testing.utils.data_gen
6. Removed test_column_ensemble_hierarchical for excessive use of overloaded operators and a poor import structure, can put it back if required
7. Removed redundant comments such as 
"capability:missing_values": False,  # can estimator handle missing data?
8. Moved a one off 3D to long converter that uses specific naming into the file where it is used

### CODE REVIEW

**Comment by TonyBagnall** [comment]
broken by refactor

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
Just one thing for now, will go through fully after other PRs

File: aeon/forecasting/base/tests/test_base.py [hunk 1]
```diff
@@ -326,35 +307,3 @@ def test_predict_residuals():
...
-
-def test_format_moving_cutoff_predictions():
-    with pytest.raises(ValueError, match="`y_preds` must be a list"):
-        _format_moving_cutoff_predictions("foo", "bar")
-    with pytest.raises(
-        ValueError, match="y_preds must be a list of pd.Series or pd.DataFrame"
-    ):
-        _format_moving_cutoff_predictions([1, 2, 3], "bar")
-    res = pd.DataFrame([1, 2, 3])
-    preds = [res, "foo"]
```
Comment (MatthewMiddlehurst) [inline aeon/forecasting/base/tests/test_base.py:339]: why is this gone?

File: aeon/forecasting/compose/tests/test_column_ensemble.py [hunk 2]
```diff
@@ -71,27 +70,3 @@ def test_column_ensemble_multivariate_and_int():
...
     fc.predict()
-
-
-@pytest.mark.skipif(
-    not _check_soft_dependencies("statsmodels", severity="none"),
-    reason="skip test if required soft dependencies not available",
-)
-def test_column_ensemble_hierarchical():
-    """Tests column ensemble with hierarchical reconciliation, see bug #3784."""
-    from aeon.datatypes import get_examples
```
Comment (MatthewMiddlehurst) [inline aeon/forecasting/compose/tests/test_column_ensemble.py:82]: keep it. I dont think its bad enough to remove yet.

File: aeon/transformations/binning.py [hunk 3]
```diff
@@ -40,16 +40,6 @@ class TimeBinAggregate(BaseTransformer):
...
-
-    Example
-    -------
-    from aeon.datatypes import get_examples
-    from aeon.transformations.binning import TimeBinAggregate
-
-    bins = [0, 2, 4]
-    X = get_examples("pd.DataFrame")[0]
-
-    t = TimeBinAggregate([-1, 2, 10])
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/binning.py:52]: whats the issue with this?

File: aeon/testing/utils/data_gen/_data_generators.py [hunk 4]
```diff
@@ -269,3 +267,123 @@ def _convert_tsf_to_hierarchical(
     df = df.astype({value_column_name: "float"}, errors="ignore")
 
     return df
+
+
+def _make_3d_test_data(
```
Comment (MatthewMiddlehurst) [inline aeon/testing/utils/data_gen/_data_generators.py:272]: is #1129 a follow up to this PR? it seems like this is going against the goal of that somewhat

File: aeon/transformations/compose.py [hunk 5]
```diff
@@ -880,7 +879,23 @@ class MultiplexTransformer(_HeterogenousMetaEstimator, _DelegatedTransformer):
...
+            "numpy3D",
+            "np-list",
+            "pd.DataFrame",
+            "pd.Series",
+            "dask_hierarchical",
+            "np.ndarray",
+            "dask_series",
+            "nested_univ",
+            "pd-wide",
+        ],
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/compose.py:898]: does the new list break this somehow?

File: aeon/transformations/compose.py [hunk 6]
```diff
@@ -917,7 +932,7 @@ def __init__(
         self.clone_tags(self.transformer_)
         self.set_tags(**{"fit_is_empty": False})
         # this ensures that we convert in the inner estimator, not in the multiplexer
-        self.set_tags(**{"X_inner_type": ALL_TIME_SERIES_TYPES})
+        self.set_tags(**{"X_inner_type": CORE_TYPES})
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/compose.py:935]: same as above

File: aeon/utils/tests/test_index_functions.py
Comment (MatthewMiddlehurst) [inline aeon/utils/tests/test_index_functions.py:1]: I dont know enough about this code to know whats going on here tbh

**Review by MatthewMiddlehurst — COMMENTED** [review]
this feels like it has some data generation stuff mixed up in it

File: aeon/utils/tests/test_index_functions.py
Comment (MatthewMiddlehurst) [reply aeon/utils/tests/test_index_functions.py:1]: not a request, i'm just gonna trust its all ok

File: aeon/transformations/binning.py (hunk 3 above)
Comment (TonyBagnall) [reply aeon/transformations/binning.py:52]: its not an example, its just a comment

File: aeon/testing/utils/data_gen/_data_generators.py (hunk 4 above)
Comment (TonyBagnall) [reply aeon/testing/utils/data_gen/_data_generators.py:272]: well these two are private and only make X, could port the code in directly but seemed pointless, just kept private functions

File: aeon/transformations/compose.py (hunk 6 above)
Comment (TonyBagnall) [reply aeon/transformations/compose.py:935]: see #1179

**Pull request merged**